tests/marine/benchmark/main.py

```diff
@@ -0,0 +1,190 @@
+
+
+def get_used_memory_in_mb() -> float:
+    return process.memory_info().rss / 1024.0 / 1024.0
+
+
+def benchmark_wrapper(f: Callable[[List[BenchmarkPacket]], None]):
+    def benchmark_timer(packets: List[BenchmarkPacket]):
+        start_used_memory = get_used_memory_in_mb()
+        start_time = time.time()
+        f(packets)
+        end_time = time.time()
+        end_used_memory = get_used_memory_in_mb()
+        delta_time = end_time - start_time
+        delta_used_memory = end_used_memory - start_used_memory
+        print(
+            f"""
+Executed {f.__name__} on {len(packets)} packets in {delta_time} seconds, 
+which is {len(packets) / delta_time} packets per second.
+Started with {start_used_memory} MB, ended with {end_used_memory}. Delta is {delta_used_memory} MB.
+"""
+        )
+
+    benchmark_functions[f.__name__] = benchmark_timer
+    return benchmark_timer
+
+
+@benchmark_wrapper
+def benchmark_bpf(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, bpf=packet.good_bpf
+        )
+        assert passed
+        assert result is None
+
+
+@benchmark_wrapper
+def benchmark_display_filter(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, display_filter=packet.good_display_filter
+        )
+        assert passed
+        assert result is None
+
+
+@benchmark_wrapper
+def benchmark_bpf_and_display_filter(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet,
+            bpf=packet.good_bpf,
+            display_filter=packet.good_display_filter,
+        )
+        assert passed
+        assert result is None
+
+
+@benchmark_wrapper
+def benchmark_3_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        fields_to_extract = packet.fields_to_extract[:3]
+        expected = dict()
+        for field in fields_to_extract:  # TODO improve this
+            expected[field] = packet.expected_parse_result[field]
+
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, fields=packet.fields_to_extract[:3]
+        )
+        assert passed
+        assert expected == result
+
+
+@benchmark_wrapper
+def benchmark_8_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        passed, result = marine_instance.filter_and_parse(
+            packet.packet, fields=packet.fields_to_extract
+        )
+        assert passed
+        assert packet.expected_parse_result == result
+
+
+@benchmark_wrapper
+def benchmark_bpf_and_display_filter_and_3_fields(packets: List[BenchmarkPacket]):
+    for packet in packets:
+        fields_to_extract = packet.fields_to_extract[:3]
+        expected = dict()
+        for field in fields_to_extract:  # TODO improve this
```

Comment:
Seems reasonable in extract this entire loop body to a function that accepts `BenchmarkPacket`.
Maybe even a method.